core/bin/oauth.rs

```diff
@@ -203,6 +204,74 @@ async fn connections_access_token(
     }
 }
 
+#[derive(Deserialize)]
+struct CredentialPayload {
+    provider: CredentialProvider,
+    raw_json: String,
+}
+
+async fn oauth_service_create_credential(
+    State(state): State<Arc<OAuthState>>,
+    Json(payload): Json<CredentialPayload>,
+) -> (StatusCode, Json<APIResponse>) {
+    match Credential::create(
+        state.store.clone(),
+        payload.provider,
+        serde_json::Value::String(payload.raw_json),
```

Comment:
same here, I would just make this param a string

Reply:
Not sure why it is being wrapped in a serde string ?

Reply:
> Not sure why it is being wrapped in a serde string ?

I don't really know either :)

Reply:
We make sure it's JSON but then consider it as a string and store it as such. If aligned with Connection logic all good 👍

Reply:
doesn’t help much to wrap in in a serde string though. Any string can be a serde string so it doesnt validate anything

Reply:
Yes this line makes no sense indeed.

Reply:
Assuming the above this should just be payload.credentials

@Fraggle not that `raw_json` is used in `oauth` (and `core`) to name a string that contains JSON but has already been serialized.

Reply:
So to sum up:

- naming: `raw_json` => `credentials`
- payload expect JSON (using `serde_json::Value`
- then turned into an encrypted string in the DB (same as today)

Did I miss something ?

Reply:
serde::Value won’t enforce json (since a simple string is a valid value)

Reply:
You mean won't enforce json `object` right ?

Reply:
Is `serde_json::Map<String, serde_json::Value>` the right type (I asked gpt4) ?

Reply:
I think its right yes 👍